code/modules/mob/living/simple_animal/hostile/wastebots.dm

```diff
@@ -56,17 +60,77 @@
 	icon_dead = "protectron_dead"
 	health = 160
 	maxHealth = 160
-	healable = 0
-	speed = 2
-	melee_damage_lower = 25
-	melee_damage_upper = 25
-	rapid = 1
-	ranged = 1
+	speed = 4
+	melee_damage_lower = 45
+	melee_damage_upper = 45
+	extra_projectiles = 2
+	ranged = TRUE
 	retreat_distance = 2
 	minimum_distance = 2
 	attacktext = "slaps"
 	attack_sound = 'sound/weapons/punch1.ogg'
 	projectilesound = 'sound/weapons/laser.ogg'
 	projectiletype = /obj/item/projectile/beam/laser/pistol
 	faction = list("wastebot")
-	check_friendly_fire = 1
+	check_friendly_fire = TRUE
+
+/mob/living/simple_animal/hostile/handy/sentrybot
+	name = "sentry bot"
+	desc = "A pre-war military robot armed with a deadly gatling laser and covered in thick armor plating."
+	icon_state = "sentrybot"
+	icon_living = "sentrybot"
+	icon_dead = "sentrybot"
+	health = 360
+	maxHealth = 360
+	melee_damage_lower = 85
+	melee_damage_upper = 85
+	extra_projectiles = 5 //6 projectiles
+	ranged_cooldown_time = 12 //brrrrrrrrrrrrt
+	ranged = TRUE
+	minimum_distance = 0
+	del_on_death = FALSE
+	attacktext = "pulverizes"
+	attack_sound = 'sound/weapons/punch1.ogg'
+	projectilesound = 'sound/weapons/laser.ogg'
+	projectiletype = /obj/item/projectile/beam/laser/pistol/weak
+	faction = list("wastebot")
+	check_friendly_fire = TRUE
+
+/obj/item/projectile/beam/laser/pistol/weak
+	damage = 10 //quantity over quality
+
+/mob/living/simple_animal/hostile/handy/sentrybot/bullet_act(obj/item/projectile/Proj)
+	if(!Proj)
+		CRASH("[src] sentrybot invoked bullet_act() without a projectile")
+		return
```

Comment:
```suggestion
```
`CRASH()` will end the proc, same as a `return`, it will just provide info and a stack trace.